packages/components/src/InputText/index.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { ForwardedRef } from "react";
+import {
+  InputText as InputTextLegacy,
+  InputTextPropOptions,
+  InputTextRef,
+} from "./InputText";
+import { InputTextSPAR } from "./InputText.rebuilt";
+import { InputTextRebuiltProps } from "./InputText.types";
+
+type CombinedProps = InputTextPropOptions | InputTextRebuiltProps;
```

Comment:
Look into conditional types because typescript suggestions are really weird at the moment
https://www.typescriptlang.org/docs/handbook/2/conditional-types.html